pytorch_lightning/metrics/utils.py

```diff
@@ -146,3 +146,35 @@ def _input_format_classification_one_hot(
         target = target.transpose(1, 0)
 
     return preds.reshape(num_classes, -1), target.reshape(num_classes, -1)
+
+
+def _input_one_hot_filtering(
+        one_hot_labels: torch.Tensor, pos_labels: List[int], inverse: bool = False
+) -> torch.Tensor:
+    """Filter out those labels that shall be considered as non-positive.
+
+    Args:
+        one_hot_labels: transposed one hot labels, shaped as (num_class * n).
+        pos_labels: a list of labels that needs to be filtered.
+        inverse: a boolean value that determined if to inverse pos_labels to select the negative parts.
+
+    Example:
+        >>> one_hot_labels = torch.tensor([
+        ...    [1, 0, 0, 1, 1, 0],
+        ...    [0, 0, 1, 0, 0, 1],
+        ...    [0, 1, 0, 0, 0, 0]
+        ... ])  # num_class * n (3 class, 6 examples)
+        >>> pos_labels = [1]
+        >>> _input_one_hot_filtering(one_hot_labels, pos_labels)
+        tensor([[1, 0, 0, 1, 1, 0],
+                [0, 0, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 0]])
```

Comment:
if I understand "Filter out" as drop non-positive, then you have it here swapped bc you show only the non-positive

Reply:
I will do a renaming to make it more clearer.